src/Altinn.App.Core/Internal/Process/ExpressionsExclusiveGateway.cs

```diff
@@ -0,0 +1,168 @@
+using System.Text;
+using System.Text.Json;
+using Altinn.App.Core.Features;
+using Altinn.App.Core.Interface;
+using Altinn.App.Core.Internal.App;
+using Altinn.App.Core.Internal.AppModel;
+using Altinn.App.Core.Internal.Expressions;
+using Altinn.App.Core.Internal.Process.Elements;
+using Altinn.App.Core.Models;
+using Altinn.App.Core.Models.Expressions;
+using Altinn.App.Core.Models.Process;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Internal.Process
+{
+    /// <summary>
+    /// Class implementing <see cref="IProcessExclusiveGateway" /> for evaluating expressions on flows connected to a gateway
+    /// </summary>
+    public class ExpressionsExclusiveGateway : IProcessExclusiveGateway
+    {
+        private readonly LayoutEvaluatorStateInitializer _layoutStateInit;
+        private readonly IAppResources _resources;
+        private readonly IAppMetadata _appMetadata;
+        private readonly IData _dataClient;
+        private readonly IAppModel _appModel;
+
+        /// <summary>
+        /// Constructor for <see cref="ExpressionsExclusiveGateway" />
+        /// </summary>
+        /// <param name="layoutEvaluatorStateInitializer"></param>
```

Comment:
Parameter documentation seems to be the only things I can find to nit pick on :-) Remove or document.